### ISSUE TITLE
False positive error about file sizes

### ISSUE THREAD

**Issue by const-ae** [Bioconductor/BiocCheck#198]
Hi,

running `BiocCheck::BiocCheck()` prints the following error:
```
* Checking individual file sizes...
    * WARNING: Package files exceed the 5MB size limit.
```

I was surprised by the error, because R CMD check does not complain about my package size. When I checked which files were too big, it appeared that [`.findLargeFiles`](https://github.com/Bioconductor/BiocCheck/blob/e87ca7a31714fed6555d980a21eb5caf3adf46cb/R/checks.R#L175) recurses over all files (including files excluded by `.gitignore` and `.Rbuildignore`):

```
Browse[2]> tibble(files, filesizes) %>% arrange(desc(filesizes)) %>% mutate(too_large = filesizes > .MAX_FILE_SIZE)
# A tibble: 13,560 × 3
   files                                                                                        filesizes too_large
   <chr>                                                                                            <dbl> <lgl>    
 1 /Users/ahlmanne/prog/r_packages/lemur/.Rproj.user/3876F4A5/ctx/ctx-61922/environment           5412950 TRUE     
 2 /Users/ahlmanne/prog/r_packages/lemur/.Rproj.user/3876F4A5/ctx/ctx-61922/environment           5412950 TRUE     
 3 /Users/ahlmanne/prog/r_packages/lemur/.Rproj.user/3876F4A5/ctx/ctx-61922/environment           5412950 TRUE     
 4 /Users/ahlmanne/prog/r_packages/lemur/.Rproj.user/3876F4A5/ctx/ctx-61922/environment           5412950 TRUE     
 5 /Users/ahlmanne/prog/r_packages/lemur/.Rproj.user/3876F4A5/ctx/ctx-61794/options               2961544 FALSE    
 6 /Users/ahlmanne/prog/r_packages/lemur/.Rproj.user/3876F4A5/ctx/ctx-61794/options               2961544 FALSE    
 7 /Users/ahlmanne/prog/r_packages/lemur/.Rproj.user/3876F4A5/ctx/ctx-61794/options               2961544 FALSE    
 8 /Users/ahlmanne/prog/r_packages/lemur/.Rproj.user/3876F4A5/ctx/ctx-61794/options               2961544 FALSE    
 9 /Users/ahlmanne/prog/r_packages/lemur/data/glioblastoma_example_data.rda                       2701955 FALSE    
10 /Users/ahlmanne/prog/r_packages/lemur/.git/objects/ad/8eb3b2d88bee3a0cf42756dc4b8d6c695f449b   2694675 FALSE    
# ℹ 13,550 more rows
# ℹ Use `print(n = ...)` to see more rows
```

Would it be possible to make the function respect the specifications in `.gitignore`?

**Comment by const-ae** [comment]
I just realized that there is a related issue https://github.com/Bioconductor/BiocCheck/issues/39. However, I think I am running into a slightly different problem, because from what I can tell, none of the files in `.Rproj.user/*` have ever been committed to git (as you can see by comparing the github version of my package: https://github.com/const-ae/lemur).

**Comment by LiNk-NY** [comment]
Hi Constantin @const-ae 

I can see the case for `.gitignore` but I am not so sure about `.Rbuildignore` because we want to minimize large files on `git.bioconductor.org` (even though those files may not be distributed when included in the `.Rbuildignore`). 

Best,
Marcel

**Comment by LiNk-NY** [comment]
I took a look and it is more time-consuming to create a filter that makes use of `.gitignore` than to use `git` itself so I used `gert` to list the files (when the package is available). Otherwise, the full list of files is used.
See the changes at https://github.com/Bioconductor/BiocCheck/commit/f88bcbf628500bb23d1b41da1c8808473e583920

**Comment by const-ae** [comment]
That's an elegant solution, thanks for fixing it so quickly.  And you're right, the important file is .gitignore and not .Rbuildignore.
I will close the issue :)
